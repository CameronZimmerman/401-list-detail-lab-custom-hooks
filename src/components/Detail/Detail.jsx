import React from 'react'
import PropTypes from 'prop-types'
import useGallery from '../../hooks/useGallery'

function Detail() {
  const {characters, loading} = useGallery()

  if(loading) return <div>Loading...</div>
  return (
    <div role="character-card">
      <p>{characters.name}</p>
      <img src={characters.photoUrl} alt={characters.name}></img>
      <p>allies: {characters.allies.join(', ')}</p>
      <p>enemies: {characters.enemies.join(', ')}</p>
      <p>affiliation: {characters.affiliation}</p>
    </div>
  )
}

export default Detail

