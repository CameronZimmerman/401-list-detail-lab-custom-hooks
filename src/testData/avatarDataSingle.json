{
  "_id": "5cf5679a915ecad153ab68f7",
  "allies": [
    "Cabbage Corp"
  ],
  "enemies": [
    "The "
  ],
  "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/2/2f/Cabbage_merchant.png/revision/latest?cb=20140112200908",
  "name": "Cabbage merchant",
  "affiliation": " Cabbage Corp Earth Kingdom"
}