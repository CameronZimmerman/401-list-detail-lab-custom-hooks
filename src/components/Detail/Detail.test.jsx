import React from 'react'
import {MemoryRouter, Route} from 'react-router-dom'
import {createMemoryHistory} from 'history'
import {render, screen} from '@testing-library/react'
import {rest} from 'msw'
import {setupServer} from 'msw/node'
import avatarDataSingle from '../../testData/avatarDataSingle.json'
import Detail from './Detail'

describe('List component', () => {
  const history = createMemoryHistory()
  history.push("/5cf5679a915ecad153ab68f7")
  const server = setupServer(
    rest.get('https://last-airbender-api.herokuapp.com/api/v1/characters/', (req, res, ctx) => {
      res(ctx.json(avatarDataSingle))
    })
  )
  beforeAll(() => server.listen())
  afterAll(() => server.close())

  it('should render a single characters', async () => {
    render(
    <MemoryRouter initialEntries={['/5cf5679a915ecad153ab68f7']}>
      <Route path='/:id'> <Detail /> </Route>
    </MemoryRouter>
      )
    screen.getByText('Loading...')
    const card = await screen.findByRole('character-card')
    expect(card).toMatchSnapshot()
  })
})