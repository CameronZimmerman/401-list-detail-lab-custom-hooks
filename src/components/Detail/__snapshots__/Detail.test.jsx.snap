// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render a single characters 1`] = `
<div
  role="character-card"
>
  <p>
    Cabbage merchant
  </p>
  <img
    alt="Cabbage merchant"
    src="https://vignette.wikia.nocookie.net/avatar/images/2/2f/Cabbage_merchant.png/revision/latest?cb=20140112200908"
  />
  <p>
    allies: 
    Cabbage Corp
  </p>
  <p>
    enemies: 
    The 
  </p>
  <p>
    affiliation: 
     Cabbage Corp Earth Kingdom
  </p>
</div>
`;
